packages/miniflare/src/plugins/core/inspector-proxy/index.ts

```diff
@@ -0,0 +1,269 @@
+		// connected. We do this by looking at the WebSocket handshake headers:
+		//
+		//  DevTools
+		//
+		// Upgrade: websocket
+		// Host: localhost:9229
+		// (from Chrome)  User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36
+		// (from Firefox) User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:109.0) Gecko/20100101 Firefox/116.0
+		// Origin: https://devtools.devprod.cloudflare.dev
+		// ...
+		//
+		//  VSCode
+		//
+		// Upgrade: websocket
+		// Host: localhost
+		// ...
+		//
+		//  WebStorm
+		//
+		// Upgrade: websocket
+		// Host: localhost:9229
+		// Origin: http://localhost:9229
+		// ...
+		//
+		// From this, we could just use the presence of a `User-Agent` header to
+		// determine if DevTools connected, but VSCode/WebStorm could very well
+		// add this in future versions. We could also look for an `Origin` header
+		// matching the hosted DevTools URL, but this would prevent preview/local
+		// versions working. Instead, we look for a browser-like `User-Agent`.
+		const userAgent = req.headers["user-agent"] ?? "";
+		const hasFileSystemAccess = !/mozilla/i.test(userAgent);
+
+		return hasFileSystemAccess;
+	}
+
+	#inspectorId = crypto.randomUUID();
+	async #handleDevToolsJsonRequest(host: string, path: string) {
+		if (path === "/json/version") {
+			return {
+				Browser: `miniflare/v${miniflareVersion}`,
+				// TODO: (someday): The DevTools protocol should match that of workerd.
+				// This could be exposed by the preview API.
+				"Protocol-Version": "1.3",
+			};
+		}
+
+		if (path === "/json" || path === "/json/list") {
+			return this.#workerRelays.map((relay) => {
+				const workerName = relay.workerName;
+				const localHost = `${host}/${workerName}`;
+				const devtoolsFrontendUrl = `https://devtools.devprod.cloudflare.dev/js_app?theme=systemPreferred&debugger=true&ws=${localHost}`;
+
+				return {
+					id: `${this.#inspectorId}-${workerName}`,
+					type: "node", // TODO: can we specify different type?
+					description: "workers",
+					webSocketDebuggerUrl: `ws://${localHost}`,
+					devtoolsFrontendUrl,
+					devtoolsFrontendUrlCompat: devtoolsFrontendUrl,
+					// Below are fields that are visible in the DevTools UI.
+					title:
+						workerName.length === 0 || this.#workerRelays.length === 1
+							? `Cloudflare Worker`
+							: `Cloudflare Worker: ${workerName}`,
+					faviconUrl: "https://workers.cloudflare.com/favicon.ico",
+					// url: "http://" + localHost, // looks unnecessary
+				};
+			});
+		}
+
+		return null;
+	}
+
+	#runtimeMessageCounter = 1e8;
+	nextCounter() {
+		return ++this.#runtimeMessageCounter;
+	}
+
+	getInspectorURL(): URL {
+		return getWebsocketURL(this.userInspectorPort);
+	}
+
+	async updateConnection(runtimeInspectorPort: number) {
+		const workerdInspectorJson = (await fetch(
+			`http://127.0.0.1:${runtimeInspectorPort}/json`
```

Comment:
Is the runtime inspector always on 127.0.0.1?

Reply:
I think so, this does suggest so:
https://github.com/cloudflare/workers-sdk/blob/5109fdd0bd5d57f22915fdf5c8114faf687cb0d2/packages/miniflare/src/index.ts#L1567